modules/core/shared/src/main/scala/eu/timepit/refined/api/RefinedTypeOps.scala

```diff
@@ -39,3 +39,10 @@ class RefinedTypeOps[FTP, T](implicit rt: RefinedType.AuxT[FTP, T]) extends Seri
   def unsafeFrom(t: T): FTP =
     rt.unsafeRefine(t)
 }
+object RefinedTypeOps {
+  class Numeric[FTP: Min: Max, T](implicit rt: RefinedType.AuxT[FTP, T])
```

Comment:
Do we want this to be a subclass of `RefinedTypeOps`, or a mixin for it? I can see arguments either way. For instance, I doubt we'll ever be able to support `MinValue` and `MaxValue` for all numeric refined types, so maybe this would be better as:

```scala
trait MaxAndMin { this: RefinedTypeOps =>
  ...
}
```

But then I don't think we can ensure that it supports Min/Max by capturing their instances when the companion object is compiled.

Reply:
I don't get the argument against the subclass approach. Why is a subclass bad if we can't support `MinValue` and `MaxValue` for all numeric refined types? Is it because of the name `Numeric`?

Reply:
Another advantage of using a `trait` here, is that we could add `MinValue` / `MaxValue` functions to companions of refined types that only have a lower or upper bound, e.g. `PosBigInt`:
```scala
trait LowerBounded[T] { 
  def minValue(implicit min: Min[T]): T = min.min
}

trait UpperBounded[T] {
  def maxValue(implicit max: Max[T]): T = max.max
}

trait Bounded extends LowerBounded with UpperBounded

//

type PosInt = Int Refined Positive

object PosInt extends RefinedTypeOps[PosInt, Int] with Bounded[PosInt]

//

type PosBigInt = BigInt Refined Positive

object PosBigInt extends RefinedTypeOps[PosBigInt, BigInt] with LowerBounded[PosBigInt]
```
Of course we could accomplish the same with classes, but that would require a little bit of boilerplate.

Reply:
The two big advantages I see of classes for this sort of thing are:
- being able to capture the `Min` and `Max` instances at companion object assembly point, so the user of that object doesn't have to think about those details and can be sure that the `MinValue` and `MaxValue` operations will work
- the `MinValue` and `MaxValue` operations can then be `val`s

Having said that, I'm worried that if we keep things as they currently are in this PR, we will have a class called 'numeric' that doesn't apply to all numeric types :-(. We could rename it to 'Bounded' but then you wouldn't be able to mix it with other such operations that we may come up with for numeric refined companion objects in the future.

So, on balance I reckon we should change this to a trait for the moment, and then add the 'Numeric' class when we can be confident that all predicates are covered. So for now we'd have:

```scala
  trait Bounded[FTP] {
    def MinValue(implicit min: Min[FTP]): FTP = min.min
    def MaxValue(implicit max: Max[FTP]): FTP = max.max
  }
```
And then in the future we could add:
```scala
  class Numeric[FTP: Min: Max, T](implicit rt: RefinedType.AuxT[FTP, T])
      extends RefinedTypeOps[FTP, T]
      with Bounded[FTP] {
    val MinValue: FTP = super.MinValue
    val MaxValue: FTP = super.MaxValue
  }
```